zebrad/src/application.rs

```diff
@@ -121,6 +141,32 @@ impl Application for ZebradApp {
         self.state.components.register(components)
     }
 
+    /// Load this application's configuration and initialize its components.
+    fn init(&mut self, command: &Self::Cmd) -> Result<(), FrameworkError> {
+        // Load configuration
+        let config = command
+            .config_path()
+            .map(|path| self.load_config(&path))
+            .transpose()?
+            .unwrap_or_default();
+
+        let config = command.process_config(config)?;
+        self.config = Some(config);
+
+        // Create and register components with the application.
+        // We do this first to calculate a proper dependency ordering before
+        // application configuration is processed
+        self.register_components(command)?;
+
+        let config = self.config.take().unwrap();
```

Comment:
Weeeeeeeeeeeeee, everyone gets a turn owning the config.

This was necessary to get access to the config when we're setting up our tracing subscriber.